io/bufio.ts

```diff
@@ -15,23 +15,36 @@ const MAX_CONSECUTIVE_EMPTY_READS = 100;
 const CR = charCode("\r");
 const LF = charCode("\n");
 
-export type BufState =
-  | null
-  | "EOF"
-  | "BufferFull"
-  | "ShortWrite"
-  | "NoProgress"
-  | Error;
+export class BufferFullError extends Error {
+  constructor(readonly partial: Uint8Array) {
+    super("Buffer full");
+  }
+}
+
+export class UnexpectedEOFError extends Error {
+  constructor() {
+    super("Unexpected EOF");
+  }
+}
```

Comment:
It would be good to have a utility for creating error classes. It’s kinda a lot of code. Also I believe you forgot to set this.name here.

Reply:
Fixed the omission of `name`.
I added 2 error classes with 5 lines of code each. Not enough to warrant a utility method.